docs/src/literate/helmholtz.jl

```diff
@@ -0,0 +1,170 @@
+# # Helmholtz equation
+#
+# In this example, we want to solve a (variant of) of the [Helmholtz equation](https://en.wikipedia.org/wiki/Helmholtz_equation).
+# The example is inspired by an [dealii step_7](https://www.dealii.org/8.4.1/doxygen/deal.II/step_7.html) on the standard square.
+#
+# ```math
+#  - \Delta u + u = f
+# ```
+#
+# With boundary conditions given by
+# ```math
+# u = g_1 \quad x \in \Gamma_1
+# ```
+# and
+# ```math
+# n \cdot \nabla u = g_2 \quad x \in \Gamma_2
+# ```
+# 
+# Here Γ₁ is the union of the top and the right boundary of the square,
+# while Γ₂ is the union of the bottom and the left boundary.
+#
+# ![](helmholtz.png)
+#
+# We will use the following weak formulation:
+# ```math
+# \int \nabla δu \cdot \nabla u d\Omega
+# + \int δu \cdot u d\Omega
+# - \int δu \cdot f d\Omega
+# + \int δu \cdot (u - g_1) d\Gamma_1
+# + \int δu \cdot (n \cdot \nabla u - g_2) d\Gamma_2
+# ```
+#
+# where $δu$ is a suitable test function.
+#
+# The example highlights the following interesting features:
+#
+# * There are two kinds of boundary conditions, "Dirichlet" and "Von Neumann"
+# * The example contains boundary integrals
+# * We will implement the von Neumann condition by hand inside the `doassemble` function.
+#
+using JuAFEM
+using Tensors
+using SparseArrays
+using LinearAlgebra
+
+const ∇ = Tensors.gradient
+const Δ = Tensors.hessian;
+
+grid = generate_grid(Quadrilateral, (150, 150))
+
+dim = 2
+ip = Lagrange{dim, RefCube, 1}()
+qr = QuadratureRule{dim, RefCube}(2)
+qr_face = QuadratureRule{dim-1, RefCube}(2)
+cellvalues = CellScalarValues(qr, ip);
+facevalues = FaceScalarValues(qr_face, ip);
+
+dh = DofHandler(grid)
+push!(dh, :u, 1)
+close!(dh)
+
+# We will set things up, so that a known analytic solution is approximately reproduced.
+# This is a good testing strategy for PDE codes and known as the method of manufactured solutions.
+
+function u_ana(x::Vec{2, T}) where {T}
+    xs = (Vec{2}((-0.5,  0.5)),
+          Vec{2}((-0.5, -0.5)),
+          Vec{2}(( 0.5,  -0.5)))
+    σ = 1/8
+    s = zero(eltype(x))
+    for i in 1:3
+        s += exp(- norm(x - xs[i])^2 / σ^2)
+    end
+    return max(1e-15 * one(T), s) # Denormals, be gone
```

Comment:
Why we need to kill denormals?

Reply:
They typically make computations very slow. But I am not sure why I did it like this and not just clamp to zero. We could remove it perhaps.